src/gamestate.rs

```diff
@@ -1,795 +0,0 @@
-        if let Tile::Piece(pawn) = self.board[x as usize][y as usize] {
-            let mut direction = if pawn.color == Color::Black { 1 } else { -1 };
-            if in_bounds((x + direction, y)) && self.is_empty((x + direction, y)) {
-                moves.insert((x + direction, y));
-            }
-            if in_bounds((x + direction, y)) && self.is_empty((x + direction, y)) && !pawn.has_moved
-            {
-                direction *= 2;
-
-                if in_bounds((x + direction, y)) && self.is_empty((x + direction, y)) {
-                    moves.insert((x + direction, y));
-                }
-            }
-
-            moves
-        } else {
-            moves
-        }
-    }
-
-    fn was_there_enemy_pawn_move_ago(&self, (x, y): Point, color: Color) -> bool {
-        self.prev_game_state.as_ref().map_or(false, |prev| {
-            if let Tile::Piece(
-                piece @ Piece {
-                    state: PieceState::Pawn,
-                    ..
-                },
-            ) = prev.board[x as usize][y as usize]
-            {
-                piece.color != color
-            } else {
-                false
-            }
-        })
-    }
-
-    fn attack_pawn_moves(&self, point @ (x, y): Point) -> HashSet<Point> {
-        let mut moves = HashSet::new();
-        if let Tile::Piece(pawn) = self.board[x as usize][y as usize] {
-            let tile_color = self.get_tile_color(point).expect("should not panic");
-            let direction = if pawn.color == Color::Black { 1 } else { -1 };
-            if in_bounds((x + direction, y + 1))
-                && self.is_enemy((x + direction, y + 1), tile_color)
-            {
-                moves.insert((x + direction, y + 1));
-            }
-
-            if in_bounds((x + direction, y - 1))
-                && self.is_enemy((x + direction, y - 1), tile_color)
-            {
-                moves.insert((x + direction, y - 1));
-            }
-
-            if self.prev_game_state.is_some() {
-                if in_bounds((x, y + 1))
-                    && self.is_enemy((x, y + 1), tile_color)
-                    && in_bounds((x + 2 * direction, y + 1))
-                    && self.was_there_enemy_pawn_move_ago((x + 2 * direction, y + 1), tile_color)
-                {
-                    moves.insert((x + direction, y + 1));
-                }
-
-                if in_bounds((x, y - 1))
-                    && self.is_enemy((x, y - 1), tile_color)
-                    && in_bounds((x + 2 * direction, y - 1))
-                    && self.was_there_enemy_pawn_move_ago((x + 2 * direction, y - 1), tile_color)
-                {
-                    moves.insert((x + direction, y - 1));
-                }
-            }
-
-            moves
-        } else {
-            moves
-        }
-    }
-
-    fn is_enemy(&self, (x, y): Point, color: Color) -> bool {
-        if let Tile::Piece(piece) = self.board[x as usize][y as usize] {
```

Comment:
don't use array indexing without doing bounds checks

Reply:
yea what i did was pretty stupid whenever i used this function i would also check its bounds but i should have just checked in the function and other similar functions inside of like putting the bounds checking right next to every innvocation